File: mobile/src/main/java/com/alexstyl/specialdates/date/DateAndTime.java
package com.alexstyl.specialdates.date;

import com.alexstyl.specialdates.TimeOfDay;

public final class DateAndTime {
    private final Date date;
    private final TimeOfDay timeOfDay;

    public DateAndTime(Date date, TimeOfDay timeOfDay) {
        this.date = date;
        this.timeOfDay = timeOfDay;
    }

    public long toMilis() {
        return date.toMillis() + timeOfDay.toMillis();
    }

    public DateAndTime addDay(int i) {
        return new DateAndTime(date.addDay(i), timeOfDay);
    }
}

File: memento/src/main/java/com/alexstyl/specialdates/date/DateAndTime.kt
package com.alexstyl.specialdates.date

import com.alexstyl.specialdates.TimeOfDay

data class DateAndTime(private val date: Date, private val timeOfDay: TimeOfDay) {

    fun toMilis(): Long {
        return date.toMillis() + timeOfDay.toMillis()
    }

    fun addDay(i: Int): DateAndTime {
        return DateAndTime(date.addDay(i), timeOfDay)
    }
}
